app/scripts/views/components/scatterPlot/scatterPlotTemplate.js

```diff
@@ -147,9 +153,14 @@
       
       $.when(iViz.getScatterData(_self))
         .then(function(_scatterData, _hasCnaFractionData, _hasMutationCountData) {
-          if ((!_hasCnaFractionData || !_hasMutationCountData) && _self.attributes.addChartBy === 'default') {
-            _self.attributes.show = false;
-            _self.$dispatch('remove-chart', _self.attributes.attr_id,  _self.attributes.group_id);//rearrange layout
+          if (!_hasCnaFractionData || !_hasMutationCountData) {
+            if (_self.attributes.addChartBy === 'default') {
+              _self.attributes.show = false;
```

Comment:
Even when added by default, the errorMessage.noData and failedToInit should be set to true as well?